include/aspect/boundary_traction/interface.h

```diff
@@ -125,7 +125,181 @@ namespace aspect
+         * plugins are implemented to register these plugins, rather than also
+         * having to modify the Manager class by adding the new boundary
+         * traction plugin class.
+         *
+         * @param name A string that identifies the boundary traction model
+         * @param description A text description of what this model does and that
+         * will be listed in the documentation of the parameter file.
+         * @param declare_parameters_function A pointer to a function that can be
+         * used to declare the parameters that this boundary traction model
+         * wants to read from input files.
+         * @param factory_function A pointer to a function that can create an
+         * object of this boundary traction model.
+         */
+        static
+        void
+        register_boundary_traction (const std::string &name,
+                                    const std::string &description,
+                                    void (*declare_parameters_function) (ParameterHandler &),
+                                    std::unique_ptr<Interface<dim>> (*factory_function) ());
+
+
+        /**
+         * Return the names of all prescribed boundary traction models currently
+         * used in the computation as specified in the input file. The function
+         * returns a map between a boundary identifier and a pair. The
+         * first part of the pair is a string that represents the prescribed
+         * traction components on this boundary (e.g. y, xz, or xyz) and the
+         * second part is a vector of strings that represent the names of
+         * boundary traction plugins for this boundary.
+         * If there are no prescribed boundary traction plugins
+         * for a particular boundary, this boundary identifier will not appear
+         * in the map.
+         */
+        const std::map<types::boundary_id, std::pair<std::string,std::vector<std::string>>> &
+        get_active_boundary_traction_names () const;
+
+        /**
+         * Return pointers to all boundary traction models
+         * currently used in the computation, as specified in the input file.
+         * The function returns a map between a boundary identifier and a vector
+         * of unique pointers that represent the names of prescribed traction
+         * boundary models for this boundary. If there are no prescribed
+         * boundary traction plugins for a particular boundary this boundary
+         * identifier will not appear in the map.
+         */
+        const std::map<types::boundary_id,std::vector<std::unique_ptr<BoundaryTraction::Interface<dim>>>> &
+        get_active_boundary_traction_conditions () const;
+
+        /**
+         * Return a list of boundary ids on which the traction is prescribed
+         * to be zero (no-slip).
+         */
+        const std::set<types::boundary_id> &
+        get_zero_boundary_traction_indicators () const;
```

Comment:
There is nothing like this for the boundary traction (a zero boundary traction plugin just returns a traction of 0, it is not in any way special). Remove this function here and in the source file.